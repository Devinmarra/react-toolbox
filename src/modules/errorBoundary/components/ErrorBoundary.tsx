import * as React from "react";
 
export interface ErrorBoundaryState {
    hasError: boolean;
}
 
export interface ErrorBoundaryProps {
    className?: string;
}
 
export class ErrorBoundary extends React.PureComponent<ErrorBoundaryProps, ErrorBoundaryState> {
    constructor(props: ErrorBoundaryProps) {
        super(props);
        this.state = { hasError: false };
    }
    public componentDidCatch() {
        this.setState({ hasError: true });
    }
 
    public render() {
        return (
            <>
                {
                    this.state.hasError ? <div className={this.props.className}>An Error Has Occurred</div>
                                        : this.props.children
                }
            </>
        )
    }
}
