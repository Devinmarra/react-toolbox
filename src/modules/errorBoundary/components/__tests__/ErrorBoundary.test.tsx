import * as React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallowToJson } from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

import { ErrorBoundary } from "../ErrorBoundary";

function WorkingComponent() {
    return <div>No Error</div>;
}
function BrokenComponent() {
    return <div>An Error Has Occurred</div>;
}

describe("<ErrorBoundary />", () => {

    it("display its children when there's no problem", () => {
        const wrapper = Enzyme.shallow(<ErrorBoundary><WorkingComponent /></ErrorBoundary>);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it("displays an error when component catches error", () => {
        const errorMsg = "An Error Has Occurred";
        const wrapper = Enzyme.shallow(<ErrorBoundary><WorkingComponent /></ErrorBoundary>);
        wrapper.setState({hasError: true}); 
        wrapper.update();
        expect(wrapper.text()).toBe(errorMsg);
    });
    describe("componentDidCatch", () => {
        const wrapper = Enzyme.shallow(<ErrorBoundary><BrokenComponent /></ErrorBoundary>);
        wrapper.instance().componentDidCatch!({ name: "test", message: "test" }, { componentStack: "test" });
        wrapper.update();
        it("sets hasError to true", () => {
            expect(wrapper.instance().state).toBeDefined();
        });
    });
});
