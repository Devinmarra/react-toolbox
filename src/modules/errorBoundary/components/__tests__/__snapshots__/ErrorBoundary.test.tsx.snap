// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> display its children when there's no problem 1`] = `<WorkingComponent />`;
